### PR TITLE
[KOGITO-2786] - Run smoke tests only when necesarry

### PR DESCRIPTION
**Pull request by Kaitou786**
See: https://issues.redhat.com/browse/KOGITO-2786

Signed-off-by: Tarun Khandelwal <tarkhand@redhat.com>

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.MD#sending-a-pull-request)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

**Comment by Kaitou786** [comment]
Jenkins, test this

**Comment by Kaitou786** [comment]
Jenkins, test this

**Comment by Kaitou786** [comment]
Jenkins, test this

**Comment by Kaitou786** [comment]
Jenkins, test this

**Comment by Kaitou786** [comment]
Jenkins, test this

**Comment by Kaitou786** [comment]
Jenkins, test this

**Comment by Kaitou786** [comment]
Jenkins, test this

**Comment by Kaitou786** [comment]
Jenkins, test this

**Comment by Kaitou786** [comment]
Jenkins, test this

**Comment by Kaitou786** [comment]
Jenkins, test this

File: Jenkinsfile [hunk 1]
```diff
@@ -1,91 +1,67 @@
-// Setup milestone to stop previous build from running when a new one is launched
-// The result would be:
-//  Build 1 runs and creates milestone 1
-//  While build 1 is running, suppose build 2 fires. It has milestone 1 and milestone 2. It passes 1, which causes build #1 to abort
+@Library('jenkins-pipeline-shared-libraries')_
 
-def buildNumber = env.BUILD_NUMBER as int
-if (buildNumber > 1) milestone(buildNumber - 1)
-milestone(buildNumber)
+def changeAuthor = env.ghprbPullAuthorLogin ?: CHANGE_AUTHOR
```
Comment (radtriste) [inline Jenkinsfile:3]: moving to RHBA ?

Was this taken from https://github.com/kiegroup/kogito-cloud-operator/pull/392/files ?

Comment (Kaitou786) [reply Jenkinsfile:3]: No, this is on the aws jenkins only, we are just changing the mechanism the same as we've on the RHBA. Multibranch doesn't have to run a job only on comment, but we've that on ghprb . So for checking out the PR we're using these params and functions on the shared library

Comment (radtriste) [reply Jenkinsfile:3]: oh I see.

File: .github/workflows/Kogito-jenkins-trigger-comment.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+name: Kogito Operator Commenter for jenkins
+on:
+  pull_request:
+    paths:
+      - 'pkg/**'
+      - 'cmd/**'
+      - 'test/**'
```
[the remark this answers is not in the document]
Comment (Kaitou786) [reply .github/workflows/Kogito-jenkins-trigger-comment.yaml:7]: Ah no, jenkins don't have the feature to have a trigger job on changes to a particular files, so the idea here is. Jenkins job is triggered only on a comment and we've this github workflow which comments on the PR when have the change, so kind of like a hybrid solution.

[the remark this answers is not in the document]
Comment (Kaitou786) [reply .github/workflows/Kogito-jenkins-trigger-comment.yaml:7]: Ah yes. I'll add that

**Comment by Kaitou786** [comment]
@ricardozanini @radtriste @sutaakar @vaibhavjainwiz do you think we shall go ahead with this one, the idea is to not run our `smoke tests` for every PR like a PR related to docs. Due to the limitation of secret sharing between the fork repos we can't have automatic comment for triggering the jenkins job, So for now the developer/contributor or the Gatekeeper like @ricardozanini or @radtriste can comment on the PR for testing the phrase would be: `Jenkins, test this` . This is can help us save a bit of resources.

**Comment by Kaitou786** [comment]
Jenkins, test this

**Comment by sutaakar** [comment]
@Kaitou786 Fine for me.

**Comment by Kaitou786** [comment]
Jenkins, test this

**Comment by Kaitou786** [comment]
Jenkins, test this

**Comment by Kaitou786** [comment]
Jenkins, test this

**Comment by Kaitou786** [comment]
The smoke tests are failing following #410,  should be fixed after #444

**Comment by radtriste** [comment]
I am good with that

**Pull request merged**